### PR TITLE
fix(manager/bundler): Allow upgrading bundler itself

### PR DESCRIPTION
**Pull request by benlangfeld**
## Changes

Updates Gemfile.lock appropriately when a dependency on bundler is bumped.

## Context

```ruby
source "https://rubygems.org"

gem "bundler", "2.5.4"
```

Prior to this change Renovate would fail reporting:

```
Could not find gem 'bundler'.
Did you mean builder?
```

With this change, Renovate is capable of bumping the dependency on Bundler itself.

Similar to https://github.com/renovatebot/renovate/pull/27451.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/bundler/artifacts.spec.ts [hunk 1]
```diff
@@ -967,6 +967,28 @@ describe('modules/manager/bundler/artifacts', () => {
         expect(res).toMatchObject([{ file: { path: 'Gemfile.lock' } }]);
         expect(execSnapshots).toMatchObject([{ cmd: 'bundler lock' }]);
       });
+
```
Comment (HonkingGoose) [inline lib/modules/manager/bundler/artifacts.spec.ts:970]: Hi @benlangfeld can you explain in the PR body text _why_ you're making this change, and link to relevant issues/discussions, and so on?

File: lib/modules/manager/bundler/artifacts.spec.ts [hunk 2]
```diff
@@ -967,6 +967,28 @@ describe('modules/manager/bundler/artifacts', () => {
...
+      it('updates the Gemfile.lock when bundler upgraded', async () => {
```
Comment (HonkingGoose) [inline lib/modules/manager/bundler/artifacts.spec.ts:971]: The test description needs a small rewrite. Would one of these work?

- updates the Gemfile.lock when updating the bundler package manager
- updates the Gemfile.lock after updating the bundler package manager

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/bundler/artifacts.spec.ts (hunk 2 above)
Comment (benlangfeld) [reply lib/modules/manager/bundler/artifacts.spec.ts:971]: I'm not sure I understand what extra information is conveyed by either of those. Anyone reading the source code of the bundler manager knows that bundler is a package manager already.

Comment (HonkingGoose) [reply lib/modules/manager/bundler/artifacts.spec.ts:971]: > I'm not sure I understand what extra information is conveyed by either of those.

I thought it was nice to mention that bundler is the package manager. But if the ones reading the code already know about it, you can leave it out. :smile: 

My point is that the tenses don't match in the description. Current situation:

- `updates` is current tense ("now")
- `upgraded` is past tense ("the past")

So it seems better to be clear about _what_ happens _when_ with a small rewrite with a connecting word like `when/after` and using the current tense. I've dropped the "package manager" part for you already. :upside_down_face: 

- updates the Gemfile.lock when updating bundler
- updates the Gemfile.lock after updating bundler

**Review by rarkins — APPROVED** [review]

**Pull request merged**